netstandard/tools/Trimming/FileNode.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Reflection.PortableExecutable;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    internal class FileNode : IIsIncluded
+    {
+        IEnumerable<FileNode> _dependencies;
+
+        internal const string NuGetPackageIdMetadata = "NuGetPackageId";
+        internal const string NuGetPackageVersionMetadata = "NuGetPackageVersion";
+        internal const string AdditionalDependenciesFileSuffix = ".dependencies";
+
+        public FileNode(ITaskItem fileItem, IDictionary<string, NuGetPackageNode> allPackages)
+        {
+            Name = fileItem.GetMetadata("Filename") + fileItem.GetMetadata("Extension");
+            OriginalItem = fileItem;
+            PackageId = fileItem.GetMetadata(NuGetPackageIdMetadata);
+            SourceFile = fileItem.GetMetadata("FullPath");
+
+            if (string.IsNullOrEmpty(PackageId))
+            {
+                PackageId = NuGetUtilities.GetPackageIdFromSourcePath(SourceFile);
```

Comment:
We've already de-duped files.  We aren't reading file information from the assets file.  The assets file is only being used for the package graph, and I plan to eventually remove that and use the items that are already lifted eventually.